spring-batch-neo4j/src/test/java/org/springframework/batch/extensions/neo4j/Neo4jItemReaderTests.java

```diff
@@ -16,69 +16,63 @@
 
 package org.springframework.batch.extensions.neo4j;
 
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.neo4j.cypherdsl.core.Cypher;
+import org.neo4j.cypherdsl.core.Node;
+import org.neo4j.cypherdsl.core.Statement;
+import org.springframework.data.neo4j.core.Neo4jTemplate;
+
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.List;
 
-import org.junit.Rule;
-import org.junit.Test;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnit;
-import org.mockito.junit.MockitoRule;
-import org.neo4j.ogm.session.Session;
-import org.neo4j.ogm.session.SessionFactory;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.ArgumentMatchers.isNull;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public class Neo4jItemReaderTests {
 
-	@Rule
-	public MockitoRule rule = MockitoJUnit.rule().silent();
+	private List<String> result;
+	private Neo4jTemplate neo4jTemplate;
 
-	@Mock
-	private Iterable<String> result;
-	@Mock
-	private SessionFactory sessionFactory;
-	@Mock
-	private Session session;
+	@SuppressWarnings("unchecked")
+	@BeforeEach
+	void setup() {
+		neo4jTemplate = mock(Neo4jTemplate.class);
+		result = mock(List.class);
```

Comment:
I propose not to mock a `java.util.List`. It is very easy to create and it does not feel like external dependency which is usually mocked.

In the test you can write a simple statement like:

```
when(neo4jTemplate.findAll(...).thenReturn(List.of(...));
```